### PR TITLE
Fix UI bugs

### LINKED ISSUE #102: Enhance tags
Blood type enums / allergy (tag) + ui

### PR DESCRIPTION
**Pull request by Al-ez**
fixes #102

### CODE REVIEW

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 1]
```diff
@@ -8,16 +8,26 @@
...
+    private static final String CSS_THEME_APPT = "-fx-font-family: \"Segoe UI\";\n"
             + "    -fx-font-size: 13px;";
 
+    private static final String CSS_THEME_BLOODTYPE = "    -fx-text-fill: white;\n" +
+            "    -fx-background-color: #d06651;\n" +
+            "    -fx-padding: 1 3 1 3;\n" +
+            "    -fx-border-radius: 2;\n" +
+            "    -fx-background-radius: 2;\n" +
+            "    -fx-font-size: 11;";
+
```
Comment (Jaynon) [inline src/main/java/seedu/address/ui/PersonCard.java]: ok but we need to migrate this to darktheme in the future

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 2]
```diff
@@ -67,27 +79,28 @@ public PersonCard(Person person, int displayedIndex) {
...
+            appointment.setStyle(CSS_THEME_APPT + " -fx-text-fill: #86ff1c;"); // Green
         } else if (person.getAppointment().hasPassed()) {
-            appointment.setStyle(CSS_THEME + " -fx-text-fill: #ff0500;"); // Red
+            appointment.setStyle(CSS_THEME_APPT + " -fx-text-fill: #ff0500;"); // Red
         } else if (person.getAppointment().hasNotPassed()) {
-            appointment.setStyle(CSS_THEME + " -fx-text-fill: #f0c44a;"); // Yellow
+            appointment.setStyle(CSS_THEME_APPT + " -fx-text-fill: #f0c44a;"); // Yellow
         } else {
-            appointment.setStyle(CSS_THEME + " -fx-text-fill: white;"); // White
+            appointment.setStyle(CSS_THEME_APPT+ " -fx-text-fill: white;"); // White
```
Comment (Jaynon) [inline src/main/java/seedu/address/ui/PersonCard.java]: looks ok

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 3]
```diff
@@ -50,6 +60,8 @@ public class PersonCard extends UiPart<Region> {
     private Label appointment;
     @FXML
     private FlowPane tags;
+    @FXML
+    private FlowPane bloodType;
```
Comment (Jaynon) [inline src/main/java/seedu/address/ui/PersonCard.java:63]: this can be removed

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 4]
```diff
@@ -25,11 +25,10 @@ public class Tag {
     public Tag(String tagName) {
         requireNonNull(tagName);
         checkArgument(isValidTagName(tagName), MESSAGE_CONSTRAINTS);
+        this.tagName = tagName;
```
Comment (JellyPenguinnn) [inline src/main/java/seedu/address/model/tag/Tag.java:28]: lgtm

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 5]
```diff
@@ -1,5 +1,7 @@
 package seedu.address.ui;
 
+import static seedu.address.model.tag.Tag.BLOOD_TYPE_PREFIX;
```
Comment (JellyPenguinnn) [inline src/main/java/seedu/address/ui/PersonCard.java:3]: lgtm

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 6]
```diff
@@ -15,9 +17,16 @@
 public class PersonCard extends UiPart<Region> {
 
     private static final String FXML = "PersonListCard.fxml";
-    private static final String CSS_THEME = "-fx-font-family: \"Segoe UI\";\n"
+    private static final String CSS_THEME_APPT = "-fx-font-family: \"Segoe UI\";\n"
             + "    -fx-font-size: 13px;";
 
+    private static final String CSS_THEME_BLOODTYPE = "    -fx-text-fill: white;\n"
```
Comment (JellyPenguinnn) [inline src/main/java/seedu/address/ui/PersonCard.java:23]: lgtm

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 7]
```diff
@@ -15,9 +17,16 @@
 public class PersonCard extends UiPart<Region> {
 
     private static final String FXML = "PersonListCard.fxml";
-    private static final String CSS_THEME = "-fx-font-family: \"Segoe UI\";\n"
+    private static final String CSS_THEME_APPT = "-fx-font-family: \"Segoe UI\";\n"
```
Comment (JellyPenguinnn) [inline src/main/java/seedu/address/ui/PersonCard.java:20]: lgtm

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 8]
```diff
@@ -50,6 +59,8 @@ public class PersonCard extends UiPart<Region> {
     private Label appointment;
     @FXML
     private FlowPane tags;
+    @FXML
```
Comment (JellyPenguinnn) [inline src/main/java/seedu/address/ui/PersonCard.java:62]: lgtm

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 9]
```diff
@@ -67,27 +78,28 @@ public PersonCard(Person person, int displayedIndex) {
         email.setText("Email : " + person.getEmail().value);
         appointment.setText("Appointment : " + person.getAppointment().dateTime);
         person.getTags().stream()
-                .sorted(Comparator.comparing(tag -> tag.tagName)) // Ensure getTagName() method exists in Tag class
+                .sorted(Comparator.comparing(tag -> tag.tagName))
```
Comment (JellyPenguinnn) [inline src/main/java/seedu/address/ui/PersonCard.java:81]: lgtm

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 10]
```diff
@@ -67,27 +78,28 @@ public PersonCard(Person person, int displayedIndex) {
...
+                        tagLabel = new Label(tag.tagName);
                     }
-
                     tags.getChildren().add(tagLabel);
+
                 });
 
         if (person.getAppointment().isToday()) {
-            appointment.setStyle(CSS_THEME + " -fx-text-fill: #86ff1c;"); // Green
+            appointment.setStyle(CSS_THEME_APPT + " -fx-text-fill: #86ff1c;"); // Green
```
Comment (JellyPenguinnn) [inline src/main/java/seedu/address/ui/PersonCard.java:96]: lgtm

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 11]
```diff
@@ -15,9 +17,16 @@
...
 
     private static final String FXML = "PersonListCard.fxml";
-    private static final String CSS_THEME = "-fx-font-family: \"Segoe UI\";\n"
+    private static final String CSS_THEME_APPT = "-fx-font-family: \"Segoe UI\";\n"
             + "    -fx-font-size: 13px;";
 
+    private static final String CSS_THEME_BLOODTYPE = "    -fx-text-fill: white;\n"
+            + "    -fx-background-color: #d06651;\n"
+            + "    -fx-padding: 1 3 1 3;\n"
+            + "    -fx-border-radius: 2;\n"
```
Comment (JellyPenguinnn) [inline src/main/java/seedu/address/ui/PersonCard.java:26]: lgtm

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 12]
```diff
@@ -25,11 +25,10 @@ public class Tag {
...
         if (ParserUtil.isValidBloodType(tagName)) {
```
Comment (Jaynon) [inline src/main/java/seedu/address/model/tag/Tag.java:29]: looks fine

File: src/main/java/seedu/address/ui/PersonCard.java (hunk 9 above)
Comment (Jaynon) [inline src/main/java/seedu/address/ui/PersonCard.java:81]: looks good

**Review by Jaynon — APPROVED** [review]
lgtm

File: src/main/java/seedu/address/ui/PersonCard.java (hunk 5 above)
Comment (colinac) [inline src/main/java/seedu/address/ui/PersonCard.java:3]: Lgtm

File: src/main/java/seedu/address/ui/PersonCard.java (hunk 6 above)
Comment (colinac) [inline src/main/java/seedu/address/ui/PersonCard.java:23]: Nice CSS

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 13]
```diff
@@ -50,6 +59,8 @@ public class PersonCard extends UiPart<Region> {
     private Label appointment;
     @FXML
     private FlowPane tags;
+    @FXML
+    private FlowPane bloodType;
```
Comment (colinac) [inline src/main/java/seedu/address/ui/PersonCard.java:63]: Adding blood type is a good addition

File: src/main/java/seedu/address/ui/PersonCard.java (hunk 9 above)
Comment (colinac) [inline src/main/java/seedu/address/ui/PersonCard.java:81]: Good use of streams

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 14]
```diff
@@ -67,27 +78,28 @@ public PersonCard(Person person, int displayedIndex) {
...
+            appointment.setStyle(CSS_THEME_APPT + " -fx-text-fill: #86ff1c;"); // Green
         } else if (person.getAppointment().hasPassed()) {
-            appointment.setStyle(CSS_THEME + " -fx-text-fill: #ff0500;"); // Red
+            appointment.setStyle(CSS_THEME_APPT + " -fx-text-fill: #ff0500;"); // Red
         } else if (person.getAppointment().hasNotPassed()) {
-            appointment.setStyle(CSS_THEME + " -fx-text-fill: #f0c44a;"); // Yellow
+            appointment.setStyle(CSS_THEME_APPT + " -fx-text-fill: #f0c44a;"); // Yellow
         } else {
-            appointment.setStyle(CSS_THEME + " -fx-text-fill: white;"); // White
+            appointment.setStyle(CSS_THEME_APPT + " -fx-text-fill: white;"); // White
```
Comment (colinac) [inline src/main/java/seedu/address/ui/PersonCard.java:102]: Lgtm

**Pull request merged**